admin/google_search_console/class-crawl-category-issues.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * @package WPSEO\Admin|Google_Search_Console
+ */
+
+/**
+ * Class WPSEO_Crawl_Category_Issues
+ */
+class WPSEO_Crawl_Category_Issues {
+
+	/**
+	 * @var string
+	 */
+	private $category;
+
+	/**
+	 * @var WPSEO_GWT_Service
+	 */
+	private $service;
+
+	/**
+	 * @var string
+	 */
+	private $platform;
+
+	/**
+	 * @var string
+	 */
+	private $option_name = '';
+
+	/**
+	 * List of all current issues to compare with received issues
+	 *
+	 * @var array
+	 */
+	private $current_issues = array();
+
+	/**
+	 * Holder for all the issues
+	 *
+	 * @var array
+	 */
+	private $issues = array();
+
+	/**
+	 * Setting up the properties and fetching the current issues
+	 *
+	 * @param string $platform
+	 * @param string $category
+	 */
+	public function __construct( $platform, $category ) {
+		$this->service     = new WPSEO_GWT_Service();
```

Comment:
Dependency injection.

You can always default to a new class using the following:

```php
public function __construct( $platform, $category, $service = null ) {

if ( null === $service ) {
    $service = new WPSEO_GWT_Service();
}
$this->service = $service;
```